src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -137,6 +138,7 @@ def __call__(
         for i, t in enumerate(self.progress_bar(self.scheduler.timesteps)):
             # expand the latents if we are doing classifier free guidance
             latent_model_input = torch.cat([latents] * 2) if do_classifier_free_guidance else latents
+            latent_model_input = latent_model_input.to(memory_format=torch.channels_last)
```

Comment:
Think this should be handled inside the model. Would be nice to keep the model input API similar.

Reply:
How much is this helping? If this gives only 5 percent speed-up not sure it's worth it as it's quite hacky IMO

Reply:
As can be found [here](https://pytorch.org/tutorials/intermediate/memory_format_tutorial.html). The gain can range from 8%-35% perf gains on Volta devices, and 26%-76% perf gains on Intel(R) Xeon(R) Ice Lake (or newer) CPUs. It also depends on the model's precision (faster for fp16 and AMP)

Reply:
But I agree that this shouldn't be proper to the stable diffusion's pipeline only. I'll try to apply one of suraj's [two suggestions](https://github.com/huggingface/diffusers/pull/371#discussion_r963630475)